src/PrestaShopBundle/Translation/TranslatorLanguageLoader.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation;
+
+use PrestaShop\PrestaShop\Core\Addon\Theme\Theme;
+use PrestaShopBundle\Translation\Loader\SqlTranslationLoader;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Translation\Loader\XliffFileLoader;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class TranslatorLanguageLoader
+{
+    /**
+     * @var bool
+     */
+    private $isAdminContext;
+
+    /**
+     * TranslatorLanguageLoader constructor.
+     *
+     * @param $isAdminContext
+     */
+    public function __construct($isAdminContext)
+    {
+        $this->isAdminContext = $isAdminContext;
+    }
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param string $locale
+     * @param Theme|null $theme
+     */
+    public function loadLanguage(TranslatorInterface $translator, $locale, Theme $theme = null)
+    {
+        if ($translator->getLocale() !== $locale) {
+            $translator->addLoader('xlf', new XliffFileLoader());
```

Comment:
Maybe you check if the loader are already present? If this method is called several times you will have multiple instances of the same loader

Reply:
From what I saw, `addLoader()` is doing `$this->loaders[$format] = $loader;` and there was no public method to get the loaders. But if you think it's better to check I can do it :)
Wdyt?

Reply:
You can also only add loaders if the catalogue is empty for that locale.